### PR TITLE
Change per-service auth annotation to use AnnotationPolicy enum.

### PR DESCRIPTION
**Pull request by diemtvu**
**What this PR does / why we need it**:
Use enum for per-service auth annotation for consistency.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

https://github.com/istio/old_auth_repo/issues/267

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pilot/model/service.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ import (
 	"fmt"
 	"sort"
 	"strings"
+
+	proxyconfig "istio.io/api/proxy/v1/config"
```
Comment (rshriram) [inline pilot/model/service.go]: @kyessenov do you think it makes sense to include proxy config here? We have kept this file independent so far.

Comment (kyessenov) [reply pilot/model/service.go]: Model can include proxy config.

Comment (diemtvu) [reply pilot/model/service.go]: So is this good to go in?

**Comment by rshriram** [comment]
sync with master

**Pull request merged**